tokyonight: Add onColor and onHightlights

dongcarl opened the pull request:
Adds the two callbacks, also adjust tests for these callbacks

GaetanLepage commented on plugins/colorschemes/tokyonight.nix:
```diff
@@ -50,6 +50,16 @@ in {
       lualineBold =
         helpers.defaultNullOpts.mkBool false
         "When true, section headers in the lualine theme will be bold";
+      onColors = mkOption {
+        type = types.nullOr helpers.rawType;
+        default = null;
+        description = "Override specific color groups to use other groups or a hex color";
+      };
+      onHighlights = mkOption {
+        type = types.nullOr helpers.rawType;
+        default = null;
+        description = "Override specific color groups to use other groups or a hex color";
+      };
```
```suggestion
      onColors = helpers.defaultNullOpts.mkStr "function(colors) end"
        ''
          Override specific color groups to use other groups or a hex color.
          function will be called with a ColorScheme table.
          `@param colors ColorScheme`
        '';
        
      onHighlights = helpers.defaultNullOpts.mkStr "function(highlights, colors) end"
        ''
          Override specific highlights to use other groups or a hex color.
          function will be called with a Highlights and ColorScheme table
          `@param highlights Highlights`
          `@param colors ColorScheme`
        '';
```

GaetanLepage commented on plugins/colorschemes/tokyonight.nix:
```diff
@@ -64,6 +74,8 @@ in {
         dim_inactive = dimInactive;
         lualine_bold = lualineBold;
         day_brightness = dayBrightness;
+        on_colors = onColors;
+        on_highlights = onHighlights;
```
```suggestion
        on_colors = helpers.ifNonNull' cfg.onColors
          (helpers.mkRaw onColors);
        on_highlights = helpers.ifNonNull' cfg.onHighlights
          (helpers.mkRaw onHighlights);
```

GaetanLepage reviewed (changes requested):
Hello !
Thank you for contributing.
Here you can make use of the helpers to declare the options.
Moreover, as those options can only be lua code, we can ask the user to simply provide a string that we will take care of converting to `rawType`.

GaetanLepage commented:
> Fixed, thanks! Would be nice to have a "hacking guide" for new contributors with "idiomatic nixvim" examples.

Thank you !
Yes indeed. For now we only have been a very few contributors, but it would surely help to have a better onboarding.

GaetanLepage commented:
I went on and rebased your branch on main. I pushed to https://github.com/GaetanLepage/nixvim/tree/2023-03-tokyonight-callbacks. Could you please apply those changes to your branch ?

dongcarl commented:
> I went on and rebased your branch on main. I pushed to https://github.com/GaetanLepage/nixvim/tree/2023-03-tokyonight-callbacks. Could you please apply those changes to your branch ?

Not sure we needed to rebase since there were no conflicts, but I rebased and pushed anyway!

GaetanLepage reviewed (approved):
LGTM

traxys commented:
@dongcarl There is a formatting issue, you can run `nix fmt` to fix it (you can run `nix flake check` to perform all checks). We should look into writing a CONTRIBUTING.md
